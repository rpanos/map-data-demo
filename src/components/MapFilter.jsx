import React from 'react'
import Form from 'react-bootstrap/Form';

import '../App.css'; // TEMP!



function MapFilter({ minSize, handleMinSize, minTot, handleMinTot }) {

  return (
    <div className="filterContainter">
      {/* <label>Limit: </label>
          <input type="text" value={limit}></input> */}
      <div className='filterField'>
        <label id="neighborhoodSizeLabel">Minimum Neighborhood Sizes:</label>
        <Form.Select
          aria-labelledby="neighborhoodSizeLabel"
          data-testid="neighborhood-size-select"
          value={minSize}
          onChange={handleMinSize}>
          <option data-testid="neighborhood-size-option" value="0">Open this select menu</option>
          <option data-testid="neighborhood-size-option" value="400000">Small</option>
          <option data-testid="neighborhood-size-option" value="800000">Medium</option>
          <option data-testid="neighborhood-size-option" value="1200000">Large</option>
          <option data-testid="neighborhood-size-option" value="1600000">Mega Large</option>
        </Form.Select>
      </div>
      <div className='filterField'>
        <label id="totalAccidentLabel">Minimum Total Accidents:</label>
        <Form.Select
          aria-labelledby="totalAccidentLabel"
          data-testid="tot-accident-select"
          value={minTot}
          onChange={handleMinTot}>
          <option data-testid="tot-accident-option" value="0">No Filter</option>
          <option data-testid="tot-accident-option" value="5">5</option>
          <option data-testid="tot-accident-option" value="10">10</option>
          <option data-testid="tot-accident-option" value="15">15</option>
          <option data-testid="tot-accident-option" value="20">20</option>
          <option data-testid="tot-accident-option" value="25">25</option>
        </Form.Select>
      </div>
    </div>
  )
}

export default MapFilter;