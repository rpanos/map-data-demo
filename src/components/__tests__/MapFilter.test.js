import MapFilter from '../MapFilter';
import {render, screen} from '@testing-library/react'
import '@testing-library/jest-dom'

const handleMinTotMock = jest.fn();
const handleMinSizeMock = jest.fn();

it("renders", () => {
    const { asFragment } = render(<MapFilter 
        handleMinTot={handleMinTotMock}
        handleMinSize={handleMinSizeMock}>
    </MapFilter>);
    expect(asFragment()).toMatchSnapshot();
});

it("will select the choice properly w/ a normal minTot", () => {
    const { asFragment } = render(<MapFilter 
        handleMinTot={handleMinTotMock}
        minTot={10}
        handleMinSize={handleMinSizeMock}>
    </MapFilter>);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('tot-accident-select')).toHaveValue("10")
});

it("will not fail if an improper size is sent in minTot", () => {
    const { asFragment } = render(<MapFilter 
        handleMinTot={handleMinTotMock}
        minTot={88}
        handleMinSize={handleMinSizeMock}>
    </MapFilter>);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('tot-accident-select')).toHaveValue("0")
});

it("will select the choice properly w/ a normal minSize", () => {
    const { asFragment } = render(<MapFilter 
        handleMinTot={handleMinTotMock}
        minSize={800000}
        handleMinSize={handleMinSizeMock}>
    </MapFilter>);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('neighborhood-size-select')).toHaveValue("800000")
});

it("will not fail if an improper size is sent in minSize", () => {
    const { asFragment } = render(<MapFilter 
        handleMinTot={handleMinTotMock}
        minSize={88}
        handleMinSize={handleMinSizeMock}>
    </MapFilter>);
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByTestId('neighborhood-size-select')).toHaveValue("0")
});

