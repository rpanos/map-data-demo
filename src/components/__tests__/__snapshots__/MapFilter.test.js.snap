// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="filterContainter"
  >
    <div
      class="filterField"
    >
      <label
        id="neighborhoodSizeLabel"
      >
        Minimum Neighborhood Sizes:
      </label>
      <select
        aria-labelledby="neighborhoodSizeLabel"
        class="form-select"
        data-testid="neighborhood-size-select"
      >
        <option
          data-testid="neighborhood-size-option"
          value="0"
        >
          Open this select menu
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="400000"
        >
          Small
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="800000"
        >
          Medium
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="1200000"
        >
          Large
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="1600000"
        >
          Mega Large
        </option>
      </select>
    </div>
    <div
      class="filterField"
    >
      <label
        id="totalAccidentLabel"
      >
        Minimum Total Accidents:
      </label>
      <select
        aria-labelledby="totalAccidentLabel"
        class="form-select"
        data-testid="tot-accident-select"
      >
        <option
          data-testid="tot-accident-option"
          value="0"
        >
          No Filter
        </option>
        <option
          data-testid="tot-accident-option"
          value="5"
        >
          5
        </option>
        <option
          data-testid="tot-accident-option"
          value="10"
        >
          10
        </option>
        <option
          data-testid="tot-accident-option"
          value="15"
        >
          15
        </option>
        <option
          data-testid="tot-accident-option"
          value="20"
        >
          20
        </option>
        <option
          data-testid="tot-accident-option"
          value="25"
        >
          25
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;

exports[`will not fail if an improper size is sent in minSize 1`] = `
<DocumentFragment>
  <div
    class="filterContainter"
  >
    <div
      class="filterField"
    >
      <label
        id="neighborhoodSizeLabel"
      >
        Minimum Neighborhood Sizes:
      </label>
      <select
        aria-labelledby="neighborhoodSizeLabel"
        class="form-select"
        data-testid="neighborhood-size-select"
      >
        <option
          data-testid="neighborhood-size-option"
          value="0"
        >
          Open this select menu
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="400000"
        >
          Small
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="800000"
        >
          Medium
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="1200000"
        >
          Large
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="1600000"
        >
          Mega Large
        </option>
      </select>
    </div>
    <div
      class="filterField"
    >
      <label
        id="totalAccidentLabel"
      >
        Minimum Total Accidents:
      </label>
      <select
        aria-labelledby="totalAccidentLabel"
        class="form-select"
        data-testid="tot-accident-select"
      >
        <option
          data-testid="tot-accident-option"
          value="0"
        >
          No Filter
        </option>
        <option
          data-testid="tot-accident-option"
          value="5"
        >
          5
        </option>
        <option
          data-testid="tot-accident-option"
          value="10"
        >
          10
        </option>
        <option
          data-testid="tot-accident-option"
          value="15"
        >
          15
        </option>
        <option
          data-testid="tot-accident-option"
          value="20"
        >
          20
        </option>
        <option
          data-testid="tot-accident-option"
          value="25"
        >
          25
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;

exports[`will not fail if an improper size is sent in minTot 1`] = `
<DocumentFragment>
  <div
    class="filterContainter"
  >
    <div
      class="filterField"
    >
      <label
        id="neighborhoodSizeLabel"
      >
        Minimum Neighborhood Sizes:
      </label>
      <select
        aria-labelledby="neighborhoodSizeLabel"
        class="form-select"
        data-testid="neighborhood-size-select"
      >
        <option
          data-testid="neighborhood-size-option"
          value="0"
        >
          Open this select menu
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="400000"
        >
          Small
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="800000"
        >
          Medium
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="1200000"
        >
          Large
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="1600000"
        >
          Mega Large
        </option>
      </select>
    </div>
    <div
      class="filterField"
    >
      <label
        id="totalAccidentLabel"
      >
        Minimum Total Accidents:
      </label>
      <select
        aria-labelledby="totalAccidentLabel"
        class="form-select"
        data-testid="tot-accident-select"
      >
        <option
          data-testid="tot-accident-option"
          value="0"
        >
          No Filter
        </option>
        <option
          data-testid="tot-accident-option"
          value="5"
        >
          5
        </option>
        <option
          data-testid="tot-accident-option"
          value="10"
        >
          10
        </option>
        <option
          data-testid="tot-accident-option"
          value="15"
        >
          15
        </option>
        <option
          data-testid="tot-accident-option"
          value="20"
        >
          20
        </option>
        <option
          data-testid="tot-accident-option"
          value="25"
        >
          25
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;

exports[`will select the choice properly w/ a normal minSize 1`] = `
<DocumentFragment>
  <div
    class="filterContainter"
  >
    <div
      class="filterField"
    >
      <label
        id="neighborhoodSizeLabel"
      >
        Minimum Neighborhood Sizes:
      </label>
      <select
        aria-labelledby="neighborhoodSizeLabel"
        class="form-select"
        data-testid="neighborhood-size-select"
      >
        <option
          data-testid="neighborhood-size-option"
          value="0"
        >
          Open this select menu
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="400000"
        >
          Small
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="800000"
        >
          Medium
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="1200000"
        >
          Large
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="1600000"
        >
          Mega Large
        </option>
      </select>
    </div>
    <div
      class="filterField"
    >
      <label
        id="totalAccidentLabel"
      >
        Minimum Total Accidents:
      </label>
      <select
        aria-labelledby="totalAccidentLabel"
        class="form-select"
        data-testid="tot-accident-select"
      >
        <option
          data-testid="tot-accident-option"
          value="0"
        >
          No Filter
        </option>
        <option
          data-testid="tot-accident-option"
          value="5"
        >
          5
        </option>
        <option
          data-testid="tot-accident-option"
          value="10"
        >
          10
        </option>
        <option
          data-testid="tot-accident-option"
          value="15"
        >
          15
        </option>
        <option
          data-testid="tot-accident-option"
          value="20"
        >
          20
        </option>
        <option
          data-testid="tot-accident-option"
          value="25"
        >
          25
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;

exports[`will select the choice properly w/ a normal minTot 1`] = `
<DocumentFragment>
  <div
    class="filterContainter"
  >
    <div
      class="filterField"
    >
      <label
        id="neighborhoodSizeLabel"
      >
        Minimum Neighborhood Sizes:
      </label>
      <select
        aria-labelledby="neighborhoodSizeLabel"
        class="form-select"
        data-testid="neighborhood-size-select"
      >
        <option
          data-testid="neighborhood-size-option"
          value="0"
        >
          Open this select menu
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="400000"
        >
          Small
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="800000"
        >
          Medium
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="1200000"
        >
          Large
        </option>
        <option
          data-testid="neighborhood-size-option"
          value="1600000"
        >
          Mega Large
        </option>
      </select>
    </div>
    <div
      class="filterField"
    >
      <label
        id="totalAccidentLabel"
      >
        Minimum Total Accidents:
      </label>
      <select
        aria-labelledby="totalAccidentLabel"
        class="form-select"
        data-testid="tot-accident-select"
      >
        <option
          data-testid="tot-accident-option"
          value="0"
        >
          No Filter
        </option>
        <option
          data-testid="tot-accident-option"
          value="5"
        >
          5
        </option>
        <option
          data-testid="tot-accident-option"
          value="10"
        >
          10
        </option>
        <option
          data-testid="tot-accident-option"
          value="15"
        >
          15
        </option>
        <option
          data-testid="tot-accident-option"
          value="20"
        >
          20
        </option>
        <option
          data-testid="tot-accident-option"
          value="25"
        >
          25
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
